### PR TITLE
Fix str_to_date return date throw error type

### PR DESCRIPTION
**Pull request by Seaven**
For #1218

The bug reason is retrun type of function `str_to_date` is changeable.
For compatible with mysql, the function need return date some times, and return datetime some times, it's let optimizer confusing.

### CODE REVIEW

File: gensrc/script/vectorized/vectorized_functions.py [hunk 1]
```diff
@@ -317,6 +317,7 @@
      'TimeFunctions::format_prepare', 'TimeFunctions::format_close'],
     [50242, 'date_format', 'VARCHAR', ['DATE', 'VARCHAR'], 'TimeFunctions::date_format',
      'TimeFunctions::format_prepare', 'TimeFunctions::format_close'],
+    [50243, 'str2date', 'DATE', ['VARCHAR', 'VARCHAR'], 'TimeFunctions::str2date', 'TimeFunctions::str_to_date_prepare', 'TimeFunctions::str_to_date_close'],
```
Comment (kangkaisen) [inline gensrc/script/vectorized/vectorized_functions.py:320]: Add  comment here

**Review by kangkaisen — DISMISSED** [review]

**Comment by Seaven** [comment]
run starrocks_clang-format

**Comment by Seaven** [comment]
run starrocks_clang-format

**Comment by wangruin** [comment]
run starrocks_clang-format

**Review by kangkaisen — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**